### DISCUSSION TITLE
how to add new protocol not so trivial

### DISCUSSION THREAD

**Discussion by neurodavidus** [wehr-lab/autopilot#82] [General]
Hello
So I wanted to run the blink task example. I found it could be useful to explain with a bit more detail the process to make this task appear in the list of tasks when we create a new protocole.
Indeed it requires to
1) copy the blink.py file in the example folder to the autopilot/autopilot/task folder
2) modify the _init_.py files and add  `from autopilot.tasks.blink import Blink`
3) modify the Task_List: 
```
TASK_LIST = {'2AFC':Nafc,
             '2AFC_Gap':Nafc_Gap,
             '2AFC_Gap_Laser':Nafc_Gap_Laser,
             'Free Water':Free_Water,
             'GoNoGo': GoNoGo,
             'Parallax': Parallax,
             'Test_DLC_Latency': DLC_Latency,
             'Test_DLC_Hand':DLC_Hand,
             'Blink':Blink}
```
There was a comment in the example 
![image](https://user-images.githubusercontent.com/4154936/108882574-d62aad80-7604-11eb-84d5-c53f86890fb7.png)

But for me it was not clear what it meant mostly because the word task completion seemed to implied that this is somethng that has to be done after running the task.

Hope this comments help and if I misunderstood something let me know. Thanks
David

**Comment by sneakers-the-rat** [comment]
Yes! Glad you figured this out. I agree this is needlessly complicated, in the future anything that inherits the `Task` class will be made available through a `TaskRegistry` like `TaskRegistry.get('task_name')`, and anything in `autopilot/plugins` will be automatically imported, so the way to do custom tasks going forward will be to just put them in the plugins directory rather than manually editing the task list which is a really inelegant way of doing it ;)

**Reply by neurodavidus** [reply]
great!

**Comment by sneakers-the-rat** [comment]
@arnefmeyer drafted a registry system (that i haven't had time to fully implement quite yet but getting there!!!), and you can see the WIP here: https://github.com/wehr-lab/autopilot/blob/registries/autopilot/utils/registry.py
